libraries/helpers.rb

```diff
@@ -16,12 +16,11 @@ def raw_checksums_to_hash
     # Downloads the raw checksums from hashicorp
     # https://releases.hashicorp.com/terraform/#{version}/#{checksum_file}
     def fetch_checksums
-      require 'uri'
       uri = URI.parse(node['terraform']['url_base'])
-      terraform_releases = "#{uri.scheme}://#{uri.host}/"
+      terraform_releases = "#{uri.scheme}://#{uri.host}"
       version = node['terraform']['version']
       checksum_file = "terraform_#{version}_SHA256SUMS"
-      path = "#{uri.path}/#{version}/#{checksum_file}"
+      path = "terraform/#{version}/#{checksum_file}"
```

Comment:
I started with gsub but it felt wrong. URI should be able to handle it in one shot, but it can get confused when url_base has part of a path in it. It seems to behave differently with different versions, so this second commit sometimes works, sometimes doesn't. I'll continue to toy with it.